File: src/main/java/net/lomeli/lomlib/util/SoundUtil.java
package net.lomeli.lomlib.util;

import net.minecraft.entity.Entity;
import net.minecraft.util.ResourceLocation;
import net.minecraft.util.SoundCategory;
import net.minecraft.util.SoundEvent;

import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

@SideOnly(Side.CLIENT)
public class SoundUtil {
    private static int size = 0;

    static {
        size = SoundEvent.REGISTRY.getKeys().size();
    }

    @SideOnly(Side.CLIENT)
    public static SoundEvent register(String name) {
        ResourceLocation loc = new ResourceLocation(name);
        SoundEvent e = new SoundEvent(loc);
        SoundEvent.REGISTRY.register(size, loc, e);
        size++;
        return e;
    }

    @SideOnly(Side.CLIENT)
    public static void playSoundAtEntity(Entity entity, SoundEvent event, SoundCategory category, float volume, float pitch) {
        if (entity != null && entity.worldObj != null)
            entity.worldObj.playSound(null, entity.posX, entity.posY, entity.posZ, event, category, volume, pitch);
    }
}

File: src/main/kotlin/net/lomeli/lomlib/util/client/SoundUtil.kt
package net.lomeli.lomlib.util.client

import net.minecraft.entity.Entity
import net.minecraft.util.ResourceLocation
import net.minecraft.util.SoundCategory
import net.minecraft.util.SoundEvent

import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly

@SideOnly(Side.CLIENT) object SoundUtil {
    private var size = 0

    init {
        size = SoundEvent.REGISTRY.keys.size
    }

    @SideOnly(Side.CLIENT) fun register(name: String): SoundEvent {
        val loc = ResourceLocation(name)
        val e = SoundEvent(loc)
        SoundEvent.REGISTRY.register(size, loc, e)
        size++
        return e
    }

    @SideOnly(Side.CLIENT) fun playSoundAtEntity(entity: Entity?, event: SoundEvent, category: SoundCategory, volume: Float, pitch: Float) {
        if (entity != null && entity.world != null)
            entity.world.playSound(null, entity.posX, entity.posY, entity.posZ, event, category, volume, pitch)
    }
}
